### ISSUE TITLE
Resize image on upload

### ISSUE THREAD

**Issue by olach** [fabianmichael/kirby-imagekit#11]
I'm using this plugin to resize uploaded images to a set max dimension:
https://github.com/schnti/kirby-image-shrink

The plugin is really small and simple. It triggers the `panel-file-upload` hook and resizes the uploaded image, and replaces the original.

```php
<?php
// Shrink large images on upload
kirby()->hook('panel.file.upload', 'shrinkImage');
kirby()->hook('panel.file.replace', 'shrinkImage');
function shrinkImage($file)
{
	$maxDimension = c::get('ka.image.shrink.maxDimension', 1000);
	try {
		if ($file->type() == 'image' and ($file->width() > $maxDimension or $file->height() > $maxDimension)) {
			// Get original file path
			$originalPath = $file->dir() . '/' . $file->filename();
			// Create a thumb and get its path
			$resized = $file->resize($maxDimension, $maxDimension);
			$resizedPath = $resized->dir() . '/' . $resized->filename();
			// Replace the original file with the resized one
			copy($resizedPath, $originalPath);
			unlink($resizedPath);
		}
	} catch (Exception $e) {
		return response::error($e->getMessage());
	}
}
```

But when ImageKit is installed, the automatic resizing does not work anymore, since ImageKit delays the actual resize of the image.

Is there a way to work around this issue? Or if you would add this as a feature to your plugin to automatically resize uploaded/replaced images?

**Comment by fabianmichael** [comment]
You can work around this bis adding the `'imagekit.lazy' => false` to the `thumb()` method. Without ImageKit, Kirby will just ignore this parameter. If ImageKit is present, thumbnails will be generated immedeiately.

**Comment by olach** [comment]
Thanks! Just a correction for others, `'imagekit.lazy' => false` is the correct option to use.

**Comment by fabianmichael** [comment]
Thanks for correcting my typo. ;-) I’ll add a notice within the readme of ImageKit.

**Comment by olach** [comment]
Just found out a similar problem. I'm using the [Dominant Color plugin](https://github.com/iandoe/kirby-dominant-color) to save the dominant color of an image in a field. I just found out that a lot of images are missing this color value. I then realized that this plugin stopped working when I activated this ImageKit plugin. 

It seems that the Dominant Color plugin must create a smaller thumbnail to work on. It cannot manage large images. Since the thumbnail is not generated directly because of ImageKit, no color can be extracted from the image.

I guess a lot of other image plugins will get similar errors when the ImageKit is used. I guess this plugin needs to have a long term solution to these problems. Maybe automatically create the thumbnail if a different plugin tries to use a thumbnail? I don't know if this is possible though.
